lib/core/cursor.js

```diff
@@ -330,6 +331,33 @@ Cursor.prototype.rewind = function() {
   }
 };
 
+Cursor.prototype._ensureInit = function(callback) {
```

Comment:
I'm contemplating not adding disconnectHandler code to this function. It seems weird to add it in when we want to get rid of it in the future.